pkg/executor/pai.go

```diff
@@ -200,7 +201,76 @@ func (s *paiExecutor) ExecuteTrain(cl *ir.TrainStmt) (e error) {
 	if e != nil {
 		return e
 	}
-	return s.submitPAITask(code, paiCmd, requirements, cl.Estimator)
+	if e = s.submitPAITask(code, paiCmd, requirements, cl.Estimator); e != nil {
+		return e
+	}
+	// download model from OSS to local cwd and save to sqlfs
+	// NOTE(typhoonzero): model in sqlfs will be used by sqlflow model zoo currently
+	// should use the model in sqlfs when predicting.
+	fmt.Printf("downloading model from %s\n", ossModelPathToSave)
+	if e = downloadOSSModel(ossModelPathToSave+"/", currProject); e != nil {
+		return e
+	}
+	m := model.New(s.Cwd, cl.OriginalSQL)
+	return m.Save(cl.Into, s.Session)
+}
+
+func downloadOSSModel(ossModelPath, project string) error {
+	bucket, err := getOSSModelBucket(project)
+	if err != nil {
+		return err
+	}
+	if !strings.HasSuffix(ossModelPath, "/") {
+		return fmt.Errorf("dir to download must end with /")
+	}
+	localDirParts := strings.Split(ossModelPath, "/")
+	localDir := localDirParts[len(localDirParts)-2] // the last char must be /
+	if err := os.Mkdir(localDir, 0755); err != nil {
+		return err
+	}
+	return downloadDirRecursive(bucket, ossModelPath, localDir+"/")
+}
+
+// downloadDirRecursive recursively download a directory on the OSS
+func downloadDirRecursive(bucket *oss.Bucket, dir, localDir string) error {
+	if err := os.MkdirAll(localDir, 0755); err != nil {
```

Comment:
Whether it will repeat with [the previous Mkdir](https://github.com/sql-machine-learning/sqlflow/pull/2563/files#diff-600ce59b276fc0a80dc0179b8c0255a1R228)?

Reply:
No, this function is called recursivly, each time it is called, the sub-directory should be created.